scripts/bozzcoin.coffee

```diff
@@ -0,0 +1,75 @@
+# Description:
+#   Tracks bozzcoins earned through exercise
+#
+# Commands: Hubot must be specifically @-mentioned.
+#   bozzcoins? - reports how many bozzcoins are in the account.
+#   did x (pullups|pushups|situps) - adds bozzcoins to the account depending on the exercise type.
+#   prata day - subtracts 10000 bozzcoins.
+
+module.exports = (robot) ->
+  # produces a summary of who has contributed bozzcoins
+  bozzcoinSummaries = () ->
+    output = "\n"
+    bozzcoinTracker = robot.brain.get("bozzcoinTracker")
+    for name of bozzcoinTracker
+      bozzcoinsContributed = bozzcoinTracker[name]
+      output += "#{name}: #{bozzcoinsContributed} :bozzcoin: contributed.\n"
+    output
+
+  convertToBozzcoin = (reps, exerciseType) ->
+    switch exerciseType
+      when "pullups" then return 10 * reps
+      when "pushups", "situps" then return 1 * reps
+      else return 0
+
+  cheatDayPrices = (cheatType) ->
+    switch cheatType
+      when "prata" then return 10000
+
+  robot.respond new RegExp("set bozzcoins to (\\d+)", "i"), (res) ->
+    if res.message.user.name is "lockheed"
+      bozzcoinBalance = Number.parseInt(res.match[1])
+      robot.brain.set "bozzcoinBalance", bozzcoinBalance
+      robot.brain.set "bozzcoinTracker", {}
+      res.send "Bozzcoin balance reset to #{bozzcoinBalance} :bozzcoin:. Tracker reset."
+
+  robot.respond /bozzcoins\?/i, (res) ->
+    bozzcoinBalance = robot.brain.get('bozzcoinBalance')
+    res.send "*#{bozzcoinBalance}* :bozzcoin:"
+    res.send bozzcoinSummaries()
+
+  robot.respond new RegExp("did (-?\\d+) (pullups|pushups|situps)", "i"), (res) ->
+    username = res.message.user.name
+    repsDone = Number.parseInt(res.match[1])
+    exerciseType = res.match[2]
+    switch exerciseType
+      when "pushups", "situps"
+        if repsDone > 60 or repsDone < -60
```

Comment:
Why is `repsDone < -60` needed?

Reply:
I think negative values are for undoing in case of error

Reply:
The message seems inaccurate if it's used for undoing.

Reply:
shouldn't the undoing be up to the person's current number of reps?

Reply:
@naomilwx good idea